src/units/store.rs

```diff
@@ -284,21 +230,19 @@ mod tests {
             assert!(store.by_hash.contains_key(hash));
         }
 
-        // Round in progress still has forker's unit
         assert_eq!(NodeCount(1), store.n_units_per_round[4]);
         assert!(store
             .by_coord
             .contains_key(&UnitCoord::new(4, NodeIndex(0))));
         assert!(store.by_hash.contains_key(&forker_hashes[4]));
 
-        // Rounds after round in progress are "free" of forker's units;
         for (round, hash) in forker_hashes[5..7].iter().enumerate() {
             let round = round as Round;
             let round = round + 5;
             let coord = UnitCoord::new(round, NodeIndex(0));
-            assert_eq!(NodeCount(0), store.n_units_per_round[round as usize]);
-            assert!(!store.by_coord.contains_key(&coord));
-            assert!(!store.by_hash.contains_key(hash));
+            assert_eq!(NodeCount(1), store.n_units_per_round[round as usize]);
+            assert!(store.by_coord.contains_key(&coord));
+            assert!(store.by_hash.contains_key(hash));
```

Comment:
As you mentioned to me outside gitub, this indeed is a silly way to fix test. Could you possibly instead check if the alert sent to `alerts_for_alerter` is correct? this would involve adding a setter for `alerts_for_alerter`, which is not the cleanest solution, but seems more reasonable.You could reuse the setter in `run_session`